libs/vkd3d/resource.c

```diff
@@ -5265,9 +5266,19 @@ static HRESULT d3d12_descriptor_heap_create_descriptor_pool(struct d3d12_descrip
             VkDescriptorPoolSize *vk_pool_size = &vk_pool_sizes[pool_count++];
             vk_pool_size->type = set_info->vk_descriptor_type;
             vk_pool_size->descriptorCount = descriptor_heap->desc.NumDescriptors;
+
+            if (set_info->vk_descriptor_type == VK_DESCRIPTOR_TYPE_STORAGE_BUFFER)
+                ssbo_pool = vk_pool_size;
         }
+
+        ssbo_count += vkd3d_popcount(set_info->flags & VKD3D_BINDLESS_SET_EXTRA_MASK);
     }
 
+    if (ssbo_count && !ssbo_pool)
+        ssbo_pool = &vk_pool_sizes[pool_count++];
+
+    ssbo_pool->descriptorCount += ssbo_count;
```

Comment:
What if ssbo_pool is NULL?

Reply:
Fixed; the code above also needs to initialize the pool which I've also fixed.